src/main/java/uk/ac/ebi/subs/agent/SamplesProcessor.java

```diff
@@ -78,19 +79,20 @@ public SamplesProcessor(RabbitMessagingTemplate rabbitMessagingTemplate, Submiss
         if (samplesWithUpdateRequirement.containsKey(true)) {
             List<Sample> samplesToUpdate = samplesWithUpdateRequirement.get(true);
 
-            List<Sample> samplesUpdated = submissionService.submit(samplesToUpdate, envelope.getJWTToken());
-            logger.info("Updated {} samples for submission {}", samplesUpdated.size(), envelope.getSubmission().getId());
-            announceSampleUpdate(submission.getId(), samplesUpdated);
-            certificates.addAll(certificatesGenerator.generateCertificates(samplesUpdated));
+            List<SampleSubmissionResponse> sampleResponseList = submissionService.submit(samplesToUpdate, envelope.getJWTToken());
+            logger.info("Updated {} samples for submission {}", sampleResponseList.size(), envelope.getSubmission().getId());
+            announceSampleUpdate(submission.getId(),
```

Comment:
I think this might send a sample message, even when the sample update got error(s) and not updated

Reply:
yes good catch